CRM/Contribute/BAO/FinancialProcessor.php

```diff
@@ -193,4 +193,52 @@ protected static function getFinancialItemAmountFromParams($params, $context, $l
     }
   }
 
+  /**
+   * Update all financial accounts entry.
+   *
+   * @param array $params
+   *   Contribution object, line item array and params for trxn.
+   *
+   * @param string $context
+   *   Update scenarios.
+   *
+   * @todo stop passing $params by reference. It is unclear the purpose of doing this &
+   * adds unpredictability.
+   *
+   */
+  public static function updateFinancialAccounts(&$params, $context = NULL) {
+    $inputParams = $params;
+    $isARefund = self::isContributionUpdateARefund($params['prevContribution']->contribution_status_id, $params['contribution']->contribution_status_id);
+
+    if ($context === 'changedAmount' || $context === 'changeFinancialType') {
+      // @todo we should stop passing $params by reference - splitting this out would be a step towards that.
+      $params['trxnParams']['total_amount'] = $params['trxnParams']['net_amount'] = ($params['total_amount'] - $params['prevContribution']->total_amount);
+    }
+
+    $trxn = CRM_Core_BAO_FinancialTrxn::create($params['trxnParams']);
+    // @todo we should stop passing $params by reference - splitting this out would be a step towards that.
+    $params['entity_id'] = $trxn->id;
+
+    $trxnIds['id'] = $params['entity_id'];
+    $previousLineItems = CRM_Price_BAO_LineItem::getLineItemsByContributionID($params['contribution']->id);
+    foreach ($params['line_item'] as $fieldId => $fields) {
+      $params = CRM_Contribute_BAO_FinancialProcessor::createFinancialItemsForLine($params, $context, $fields, $previousLineItems, $inputParams, $isARefund, $trxnIds, $fieldId);
+    }
+  }
+
+  /**
+   * Does this contribution status update represent a refund.
+   *
+   * @param int $previousContributionStatusID
+   * @param int $currentContributionStatusID
+   *
+   * @return bool
+   */
+  public static function isContributionUpdateARefund($previousContributionStatusID, $currentContributionStatusID): bool {
```

Comment:
public for now as still accessed by the BAO_Contribution class - in the medium term that function will move over & this can be private again